src/peer_map.rs

```diff
@@ -222,112 +142,52 @@ mod tests {
+        assert!(peer_map.get_public_id(&conn_info.peer_addr()).is_none());
 
+        peer_map.handle_direct_message(pub_id, conn_info.peer_addr());
+        assert_eq!(peer_map.get_connection_info(&pub_id), Some(&conn_info));
         assert_eq!(
-            peer_map.handle_direct_message(pub_id, socket_addr),
-            Ok(conn_info)
-        );
-        assert_eq!(
-            peer_map.handle_direct_message(pub_id, socket_addr),
-            Err(ConnectionError::AlreadyConnected)
-        );
-        assert_eq!(
-            peer_map.handle_direct_message(pub_id, socket_addr),
-            Err(ConnectionError::AlreadyConnected)
+            peer_map.get_public_id(&conn_info.peer_addr()),
+            Some(&pub_id)
         );
-    }
-
-    #[test]
-    fn connected_to_then_connection_failure() {
-        let mut peer_map = PeerMap::new();
-
-        let conn_info = conn_info("198.51.100.0:5555");
-        let socket_addr = conn_info.peer_addr();
 
         assert_eq!(
-            peer_map.handle_connected_to(conn_info),
-            Err(ConnectionError::Incomplete)
+            peer_map.handle_connection_failure(conn_info.peer_addr()),
+            Some(pub_id)
         );
-        assert!(peer_map.handle_connection_failure(socket_addr).is_none());
-    }
-
-    #[test]
-    #[should_panic(expected = "DirectMessage from peer not connected at the network layer")]
-    fn direct_message_only() {
-        let mut peer_map = PeerMap::new();
-
-        let conn_info = conn_info("198.51.100.0:5555");
-        let socket_addr = conn_info.peer_addr();
-        let pub_id = *FullId::new().public_id();
-
-        let _ = peer_map.handle_direct_message(pub_id, socket_addr);
+        assert!(peer_map.get_connection_info(&pub_id).is_none());
+        assert!(peer_map.get_public_id(&conn_info.peer_addr()).is_none());
     }
 
     #[test]
-    fn connect_request_then_connected_to() {
+    fn connect_request() {
         let mut peer_map = PeerMap::new();
-
         let node_info = node_info("198.51.100.0:5555");
         let pub_id = *FullId::new().public_id();
 
-        assert_eq!(
-            peer_map.handle_connect_request(pub_id, node_info.clone()),
-            Err(ConnectionError::Incomplete)
-        );
-        assert_eq!(
-            peer_map.handle_connected_to(ConnectionInfo::Node { node_info }),
-            Ok(pub_id)
-        );
-    }
-
-    #[test]
-    fn connected_to_then_connect_request() {
-        let mut peer_map = PeerMap::new();
-
-        let node_info = node_info("198.51.100.0:5555");
-        let pub_id = *FullId::new().public_id();
+        assert!(peer_map.get_connection_info(&pub_id).is_none());
+        assert!(peer_map.get_public_id(&node_info.peer_addr).is_none());
 
+        peer_map.handle_connect_request(pub_id, node_info.clone());
         assert_eq!(
-            peer_map.handle_connected_to(ConnectionInfo::Node {
+            peer_map.get_connection_info(&pub_id),
+            Some(&ConnectionInfo::Node {
```

Comment:
Would it be more readable if we had
`let conn_info = conn_info("198.51.100.0:5555");` under `let node_info...` so we can use it in this assert?